mypyc/test-data/analysis.test

```diff
@@ -197,31 +241,63 @@ def f(n: int) -> None:
 def f(n):
     n :: int
     r0 :: bool
-    r1, m :: int
-    r2 :: None
+    r1 :: native_int
+    r2 :: bool
+    r3 :: native_int
+    r4, r5, r6, r7 :: bool
+    r8, m :: int
+    r9 :: None
 L0:
 L1:
-    r0 = CPyTagged_IsLt(n, 10)
-    if r0 goto L2 else goto L3 :: bool
+    r1 = n & 1
+    r2 = r1 == 0
+    r3 = 10 & 1
+    r4 = r3 == 0
+    r5 = r2 & r4
+    if r5 goto L2 else goto L3 :: bool
```

Comment:
Not directly related to this diff: we could simplify this sequence of ops like this:

```
r1 = n | 10
r2 = r1 & 1
r3 = r2 == 0
if r3 goto ...
```
```